orga/tests/integration/components/auth/register-form_test.js

```diff
@@ -54,6 +54,31 @@ module('Integration | Component | Auth::RegisterForm', function (hooks) {
     assert.dom(screen.getByText('Tous les champs sont obligatoires.')).exists();
   });
 
+  test('it should display legal mentions with related links', async function (assert) {
+    // given
+    const service = this.owner.lookup('service:url');
+    service.currentDomain = { getExtension: sinon.stub().returns('fr') };
+
+    // when
+    const screen = await renderScreen(hbs`<Auth::RegisterForm/>`);
+
+    // then
+    assert.dom(screen.getByText('politique de confidentialité')).exists();
+    assert.dom(screen.getByText("conditions d'utilisation de Pix")).exists();
+    assert.dom(screen.getByText(/J'accepte/i)).exists();
```

Comment:
On avait l'erreur `Unable to find an element with the text: 2-3. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.` 

Voir l'exemple donné [ici](https://timdeschryver.dev/blog/making-sure-youre-using-the-correct-query)

> The *ByText and *ByLabelText can't find elements that are broken into multiple elements. For example, given the following HTML, it wasn't possible to query for the text "Hello World".
`<h3>Hello <span>World</span></h3>`
L'auteur préconise d'utiliser le getByRole mais ça n'avait pas trop de sens de mettre un rôle au span qui contient le texte et les liens.

Ça discute de cette erreur ici https://github.com/testing-library/dom-testing-library/issues/410
J'ai utilisé la syntaxe `/text/i` trouvé dans l'[api de testing library](https://testing-library.com/docs/queries/bytext) 

Par contre, faire `/`${this.intl.t('pages.login-or-register.register-form.fields.cgu.terms-of-use')}`/i` ça passe pas alors j'ai mis le texte en dur

Reply:
Ah je ne connaissais pas le /text/i, intéressant. Je ne trouvais pas sur ton lien mais je l'ai trouvé ici : 
https://testing-library.com/docs/queries/about/#textmatch-examples

<img width="928" alt="Capture d’écran 2022-02-07 à 11 21 41" src="https://user-images.githubusercontent.com/58915422/152769824-657643bc-a118-4af8-a231-95ec56e13d94.png">

Reply:
Donc on pourrait faire aussi bien : 
assert.dom(screen.getByText("J'accepte les", { exact: false })).exists();
OU
assert.dom(screen.getByText(/J'accepte les/)).exists();

A choisir lequel est le plus lisible. Si on garde ta version, le ` i ` n'est pas nécessaire, il ne fait qu'ignorer le case.

Reply:
Du coup, ça ça fonctionne ! 

```
assert
      .dom(
        screen.getByText(this.intl.t('pages.login-or-register.register-form.fields.cgu.terms-of-use'), { exact: false })
      )
      .exists();
```

Reply:
J'ai pas de préférences entre les propositions :) je voulais juste tester de mon coté, c'est intéressant d'avoir ça sous la main pour les prochaines fois.

Reply:
Cool ! Je vais partir sur 
```
assert
      .dom(
        screen.getByText(this.intl.t('pages.login-or-register.register-form.fields.cgu.terms-of-use'), { exact: false })
      )
      .exists();
```
comme ça on reste sur quelque chose de consistant